WordPress/gutenberg
# Quote: fix raw transform handler

## What?

Use the correct raw handler depending on the context.

## Why?

Currently we're using the `rawHandler`, which is letting too much markup through on paste.

## How?

We have two raw handlers: a default one (used for legacy context for example) and one for paste.

So far we didn't have the use case for nested raw handlers, so we never ran into this issue before.

We need to use the correct handler in the raw transform depending on which one is running. To solve that we can pass down the handler to the transform callback.

## Testing Instructions

See integration test. Diff:

```
      
      <blockquote class="wp-block-quote">
    - <p>Test with&nbsp;<a target="_blank" href="http://w.org/" rel="noreferrer noopener">link</a>.</p>
    + <p>Test with<span>&nbsp;</span><a target="_blank" class="c-link" data-stringify-link="http://w.org" delay="150" data-sk="tooltip_parent" href="http://w.org/" rel="noreferrer noopener" tabindex="-1" data-remove-tab-index="true" style="box-sizing: inherit; color: rgba(var(--sk_highlight,18,100,163),1); text-decoration: none;">link</a>.</p>
      </blockquote>
      
```

## Screenshots or screencast
